demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -624,6 +625,15 @@ CK_RV xProvisionCertificate( CK_SESSION_HANDLE xSession,
         xCertificateTemplate.xValue.ulValueLen = xDerLen;
     }
 
+    /* Best effort clean-up of the existing object, if it exists. */
+    if( xResult == CKR_OK )
+    {
+        xDestroyProvidedObjects( xSession,
```

Comment:
I'm not sure if I'm a fan of automatically destroying a matching credential within xProvisionCertificate.

Reply:
This also strikes me as non-obvious behavior.